Lib/imaplib.py

```diff
@@ -318,9 +318,11 @@ def shutdown(self):
         self.file.close()
         try:
             self.sock.shutdown(socket.SHUT_RDWR)
-        except OSError as e:
-            # The server might already have closed the connection
-            if e.errno != errno.ENOTCONN:
+        except OSError as exc:
+            # The server might already have closed the connection.
+            # Windows error 10022: An invalid operation was attempted.
```

Comment:
I would change the comment to "On Windows, this may result in Windows error 10022: An invalid operation was attempted".  At least, I hope that's why we're getting that error.  (In my experience invalid operation is indeed what windows raises when you attempt an operation on a closed socket).